test/web-platform-tests/run-web-platform-test.js

```diff
@@ -1,73 +1,95 @@
-      } else {
-        resource.defaultFetch(callback);
-      }
-    },
-    created(err, window) {
-      if (err) {
-        t.ifError(err, "window should be created without error");
-        t.done();
-        return;
-      }
+  let created;
+  const loaded = nodeResolverPromise(loadedResolver => {
+    created = nodeResolverPromise(createdResolver => {
+      jsdom.env({
+        url: urlPrefix + testPath,
+        pool: globalPool,
+        features: {
+          FetchExternalResources: ["script", "frame", "iframe", "link"],
+          ProcessExternalResources: ["script"]
+        },
+        virtualConsole: jsdom.createVirtualConsole().sendTo(console, { omitJsdomErrors: true }),
+        resourceLoader(resource, callback) {
+          if (resource.url.href === reporterHref) {
+            callback(null, "window.shimTest();");
+          } else {
+            resource.defaultFetch(callback);
+          }
+        },
+        created: createdResolver,
+        loaded: loadedResolver
+      });
+    });
+  });
+
+  return created
+  .then(window => {
+    return new Promise((resolve, reject) => {
+      const errors = [];
 
       window.shimTest = () => {
         window.add_result_callback(test => {
           if (test.status === 1) {
-            t.ok(false, "Failed in \"" + test.name + "\": \n" + test.message + "\n\n" + test.stack);
+            errors.push(`Failed in "${test.name}": \n${test.message}\n\n${test.stack}`);
           } else if (test.status === 2) {
-            t.ok(false, "Timout in \"" + test.name + "\": \n" + test.message + "\n\n" + test.stack);
+            errors.push(`Timeout in "${test.name}": \n${test.message}\n\n${test.stack}`);
           } else if (test.status === 3) {
-            t.ok(false, "Uncompleted test in \"" + test.name + "\": \n" + test.message + "\n\n" + test.stack);
-          } else {
-            t.ok(true, test.name);
+            errors.push(`Uncompleted test "${test.name}": \n${test.message}\n\n${test.stack}`);
           }
         });
 
         window.add_completion_callback((tests, harnessStatus) => {
-          t.ok(harnessStatus.status !== 2, "test harness should not timeout: " + testPath);
-
           // This needs to be delayed since some tests do things even after calling done().
           process.nextTick(() => {
             window.close();
           });
 
-          t.done();
+          if (harnessStatus.status === 2) {
+            reject(new Error(`test harness should not timeout: ${testPath}`));
+            return;
+          }
+
+          if (errors.length === 1) {
+            reject(new Error(errors[0]));
+          } else if (errors.length) {
+            reject(new Error(`${errors.length} errors in test:\n\n${errors.join("\n")}`));
+          } else {
+            resolve();
+          }
         });
       };
-    },
-
-    loaded(err) {
-      t.ifError(err);
-    }
-  });
+    });
+  })
+  .then(loaded); // make sure that the loaded callback does not error
```

Comment:
Also this callback does not even exist anymore in jsdom, the proper one is "onload". There is also a small issue in the readme that i'll fix.